dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp

```diff
@@ -222,22 +223,37 @@ DecodedLockCFValuePtr RegionData::getLockInfo(const RegionLockReadQuery & query)
         const auto & [tikv_key, tikv_val, lock_info_ptr] = value;
         std::ignore = tikv_key;
         std::ignore = tikv_val;
-        const auto & lock_info = *lock_info_ptr;
+        const auto & lock_info_raw = *lock_info_ptr;
 
-        if (lock_info.lock_version > query.read_tso || lock_info.lock_type == kvrpcpb::Op::Lock
-            || lock_info.lock_type == kvrpcpb::Op::PessimisticLock)
-            continue;
-        if (lock_info.min_commit_ts > query.read_tso)
-            continue;
-        if (query.bypass_lock_ts)
-        {
-            if (query.bypass_lock_ts->count(lock_info.lock_version))
+        bool should_continue = false;
+        lock_info_raw.withInner([&](const RecordKVFormat::DecodedLockCFValue::Inner & in) {
+            if (in.lock_version > query.read_tso || in.lock_type == kvrpcpb::Op::Lock
+                || in.lock_type == kvrpcpb::Op::PessimisticLock)
+            {
+                should_continue = true;
+                return;
+            }
+            if (in.min_commit_ts > query.read_tso)
             {
-                GET_METRIC(tiflash_raft_read_index_events_count, type_bypass_lock).Increment();
-                continue;
+                should_continue = true;
+                return;
```

Comment:
`bool withInner(std::function<bool(const Inner &)> f) const;` may be better?

Reply:
Similar, I can move it into the DecodedLockCfValue